### PR TITLE
fix(transactions): keep filters on pagination and reset on type switch

### PR DESCRIPTION
**Pull request by iguliaev**
## Why
Transactions filters were being lost during pagination and users could hit stale pagination/filter state when switching transaction type, leading to invalid list requests (including 416 range errors) and broken UI feedback.

## What Changed
- Files or areas updated:
  - `apps/web-next/src/pages/transactions/list.tsx`
  - `apps/web-next/e2e/tests/transactions.spec.ts`
- New functionality added:
  - Added e2e coverage for category-filter persistence across pagination.
  - Added e2e coverage for clearing active filters and resetting page on transaction type switch.
- Existing behavior changed:
  - Table filters (category/amount/tags/bank account) are controlled via `filteredValue` to persist correctly.
  - Transaction type now derives from synced table filters.
  - Type switch resets to page 1 and replaces filters with only the selected type.

## Key Decisions
- Decision:
  - Use `useTable` synced filter state as the single source of truth for transaction type and table filters.
- Reasoning:
  - Prevents drift between local UI state and URL/query state while keeping pagination/filter updates consistent.
- Alternatives considered:
  - Local `useState` for type with permanent filter updates; rejected due to race/state desync risk with synced pagination/filter params.

## How This Affects the System
- User-facing changes:
  - Filters persist when paginating.
  - Switching Spend/Earn/Save clears incompatible filters and resets pagination safely.
- API changes:
  - None.
- Performance implications:
  - No material impact.
- Database changes:
  - None.
- Breaking changes:
  - None.

## Testing
- New tests added:
  - `Transactions › category filter persists when moving to page 2`
  - `Transactions › switching transaction type clears active list filters`
- Existing tests status:
  - Related amount-range filter test still passes.
- Manual testing performed:
  - Through Playwright flow during debugging.
- Edge cases tested:
  - High page index + filtered list + type switch state transition.
- Test files modified or added:
  - Modified `apps/web-next/e2e/tests/transactions.spec.ts`

### CODE REVIEW

File: apps/web-next/e2e/tests/transactions.spec.ts [hunk 1]
```diff
@@ -613,6 +613,250 @@ test.describe("Transactions", () => {
...
+      await expect(page).toHaveURL(/category_id/);
+      await expect(
+        page.getByText(new RegExp(`persist-groceries-${ts}-`)).first()
+      ).toBeVisible();
+      await expect(page.getByText(otherNote)).not.toBeVisible();
+    } finally {
+      if (otherCategoryId) {
+        await supabaseAdmin.from("categories").delete().eq("id", otherCategoryId);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (iguliaev) [reply apps/web-next/e2e/tests/transactions.spec.ts:748]: Fixed in f180e210. The test now deletes seeded transactions in finally (scoped by user + notes pattern/exact note) before deleting the temporary category, preventing FK-order failures and leakage.
